src/service.js

```diff
@@ -205,65 +146,58 @@ function getLocalExistingServiceForEmbedWinOrNull(embedWin, id) {
  */
 export function getService(win, id, opt_factory) {
   win = getTopWindow(win);
-  return getServiceInternal(win, win, id, undefined,
-      opt_factory ? opt_factory : undefined);
-}
-
-
-/**
- * Returns a service and registers it given a class to be used as
- * implementation.
- * @param {!Window} win
- * @param {string} id of the service.
- * @param {function(new:T, !Window)} constructor
- * @return {T}
- * @template T
- */
-export function fromClass(win, id, constructor) {
-  win = getTopWindow(win);
-  return getServiceInternal(win, win, id, constructor);
+  if (!isServiceRegistered(win, id)) {
```

Comment:
The existence of `opt_factory` here and in `getServiceForDoc` breaks the registration pattern. What do you think we should do about this?